inc/auth.class.php

```diff
@@ -763,10 +767,10 @@ function Login($login_name, $login_password, $noauto=false) {
                $input = Toolbox::addslashes_deep($input);
 
                // update user and Blank PWD to clean old database for the external auth
-               $this->user->update($input);
                if ($this->extauth && $this->user->fields["authtype"] != self::API) {
-                  $this->user->blankPassword();
+                  $input['_extauth'] = 1;
```

Comment:
Not enough, _extauth is not manager in User::prepareInputForUpdate, see User::prepareInputForAdd)